### ISSUE TITLE
Protocols don't support descriptors

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#5481]
Consider this code:
```python
class Descr:
    def __get__(self, inst: Any, owner: Any) -> int: ...

class Proto(Protocol):
    x: int

class C:
    x = Descr()

a: Proto = C()
```
I think it should be allowed. This can be done by simply re-using the logic in `analyze_member_access` in `checkmember.py`, see also https://github.com/python/mypy/issues/5136.

**Comment by bryanforbes** [comment]
This will be fixed when #3832 is complete

**Comment by ilevkivskyi** [comment]
This appeared again in https://github.com/dropbox/sqlalchemy-stubs/issues/134

**Comment by ckarnell** [comment]
Can you tell me exactly where in the code the duplication is happening WRT Protocols, so I can try to piece together some understanding and look into fixing it? I'm looking myself but it's not immediately obvious, and I'm unfamiliar with the codebase.

**Comment by ilevkivskyi** [comment]
Just to set realistic expectations: it is unlikely anyone beyond core team can _fix_ this. But anyway, the problematic logic duplication is between `is_protocol_implementation()/find_member()` in `subtypes.py` and `analyze_instance_member_access()` in `checkmember.py`.

**Comment by ckarnell** [comment]
Got it, makes sense. I might take a peek anyway just out of curiosity.

**Comment by sobolevn** [comment]
This was raised once again in `django-stubs` repo: https://github.com/typeddjango/django-stubs/issues/651
